astropy/io/fits/hdu/hdulist.py

```diff
@@ -694,14 +694,13 @@ def writeto(self, fileobj, output_verify='exception', overwrite=False,
         # but only if the file doesn't exist.
         fileobj = _File(fileobj, mode='ostream', overwrite=overwrite)
         hdulist = self.fromfile(fileobj)
+        dirname = os.path.dirname(hdulist._file.name)
```

Comment:
I assumed that as the hdulist was just created from a a file, this should be fine. Tests pass. Would you prefer to have it in a try/except? I'm sure I'm not able to consider all possible corner case here.